lib/internal/fs/promises.js

```diff
@@ -739,6 +747,74 @@ async function readFile(path, options) {
   return PromisePrototypeFinally(readFileHandle(fd, options), fd.close);
 }
 
+async function stream(handle, dest, options = {}) {
+  validateObject(options, 'options');
+
+  if (!isWritable(dest)) {
+    throw new ERR_INVALID_ARG_TYPE('dest', 'Writable', dest);
+  }
+
+  validateAbortSignal(options.signal);
+  checkAborted(options.signal);
+  if (!EE) {
+    EE = require('events');
+  }
+  const BUFFER_SIZE = 128 * 2 ** 10;
+  const {
+    start = handle[kHandle].bytesRead,
+    end,
+  } = options;
+
+  if (start !== undefined) {
+    validateInteger(start, 'start', 0);
+  }
+  if (end !== undefined) {
+    validateInteger(end, 'end', start);
+  }
+
+  let currPos = start;
+
+  while (true) {
+    if (end && currPos > end) {
+      break;
+    }
+
+    checkAborted(options.signal);
+    if (dest.writableNeedDrain) {
+      await EE.once(dest, 'drain');
+    }
+
+    let buffer = Buffer.allocUnsafe(
+      end ? MathMin(end + 1 - currPos, BUFFER_SIZE) : BUFFER_SIZE
+    );
+
+    const { bytesRead } = await handle.read(
+      buffer,
+      0,
+      buffer.byteLength,
+      currPos
+    );
+
+    if (dest.destroyed) {
+      throw new ERR_STREAM_PREMATURE_CLOSE();
+    }
+
+    if (bytesRead === 0) {
+      break;
+    }
+
+    if (bytesRead !== buffer.length) {
+      // Shrink to fit bytes read
+      const copy = Buffer.allocUnsafeSlow(bytesRead);
+      buffer.copy(copy, 0, 0, bytesRead);
+      buffer = copy;
+    }
+
+    dest.write(buffer);
```

Comment:
```suggestion
    if (!dest.write(buffer)) {
       if (dest.destroyed) return;
       await EE.once(dest, 'drain');
    }
```

Better compat with old streams.

Reply:
Added